backend/src/packages/auth/auth.controller.ts

```diff
@@ -67,6 +67,13 @@ class AuthController extends Controller {
       body: UserSignUpRequestDto;
     }>,
   ): Promise<ApiHandlerResponse> {
+    if (await this.authService.checkExistingUser(options.body)) {
+      return {
+        status: HttpCode.CONFLICT,
+        payload: { error: 'User already exists' },
+      };
+    }
+
```

Comment:
it's part of business logic. should not be placed on controllers layer. Also create enum for error messages